java/sage/MediaFile.java

```diff
@@ -5486,6 +5489,29 @@ public String execVideoThumbGen(String pathString, boolean deinterlace, float of
       int targetStream)
   {
     List<String> args = new ArrayList<String>(25);
+    String res;
+    synchronized (this)
+    {
+      if (ffmpegSupportsVF == null || ffmpegSupportsSkipNonKeys == null)
+      {
+        // Check if FFMPEG supports -vf and the preferred -skip_ options because
+        // several versions were distributed and some do not support these features
+        args.add(Sage.EMBEDDED ? "./ffmpegjpeg" : FFMPEGTranscoder.getTranscoderPath());
+        args.add("-h");
+        res = IOUtils.exec(args.toArray(Pooler.EMPTY_STRING_ARRAY), true, true, true);
+        ffmpegSupportsVF = res.contains("-vf ");
+        ffmpegSupportsSkipNonKeys = (res.contains("-skip_frame ") && res.contains("-skip_idct ") && res.contains("-skip_loop_filter "));
+        if (Sage.DBG) System.out.println("FFMPEG thumbnail option check: ffmpegSupportsVF=" + String.valueOf(ffmpegSupportsVF) + " ffmpegSupportsSkipNonKeys=" + String.valueOf(ffmpegSupportsSkipNonKeys));
+        args.clear();
+      }
+    }
+    // Return if skipNonKeys is specified but not supported by FFMPEG, instead
+    // of executing it with invalid options
+    if (skipNonKeys && !ffmpegSupportsSkipNonKeys)
+    {
+      res="skipNonKeys options not supported by ffmpeg";
```

Comment:
I wouldn't bail out here in this case. Just don't set the options for skipping key frames...IIRC, that was basically just a performance optimization and nothing more.